controllers/ebpf/agent_controller.go

```diff
@@ -134,7 +142,10 @@ func (c *AgentController) Reconcile(ctx context.Context, target *flowslatest.Flo
 	if err != nil {
 		return err
 	}
-
+	err = c.reconcilePrometheusService(ctx, &target.Spec.Agent.EBPF)
```

Comment:
there's not only the service to reconcile here, but also the agent pod config (ENV) to pass the configured server port.
I would also add a config to enable metric
And I think we should set the metrics prefix to "netobserv_agent_"

Reply:
yeah was holding on that piece till I get the promo pieces in I added that it now